logstash-core/lib/logstash/modules/settings_merger.rb

```diff
@@ -20,4 +20,25 @@ def self.merge(cli_settings, yml_settings)
     end
     merged
   end
+
+  def self.merge_cloud_settings(module_settings, logstash_settings)
```

Comment:
Can we have a unit test that test the merging strategies, to make sure we have the correct priorities.